src/tooltip/src/Tooltip.js

```diff
@@ -22,7 +22,7 @@ const Tooltip = memo(function Tooltip(props) {
     statelessProps = emptyProps
   } = props
 
-  const id = useId('evergreen-tooltip')
+  const id = useId('evergreen-tooltip', props.id)
```

Comment:
I pulled this down and it seems to solve the warning in the original issue (when providing an explicit id). However, I think this is somewhat of a workaround - ideally, the `useId` hook should be reworked to return stable ids between the client <> server so an explicit id isn't required to suppress this warning during SSR.

I'm cool w/ adding in the ability to set an explicit id for the Tooltip either way, but I think we'll need to add it to `TooltipProps` in the `index.d.ts` file for TypeScript users:

https://github.com/segmentio/evergreen/blob/0c36c459c6964303d9877fc67b2a0c918c011170/index.d.ts#L3109-L3142

Reply:
`index.d.ts` _"should"_ be generated, but we're still on JS 🥲

>Without a unique reference for this component between the server and client, how would you create a stable id?

I'm not entirely sure! I'm not super familiar with SSR myself, but I did some quick searching and it looks like `useId` is available as a hook directly from React in v18 which solves this: https://github.com/reactwg/react-18/discussions/111

Mantine implements a [stable id hook](https://github.com/mantinedev/mantine/blob/master/src/mantine-hooks/src/use-id/use-id.ts) by setting the state in an effect: 

```tsx
import React, { useState } from 'react';
import { useIsomorphicEffect } from '../use-isomorphic-effect/use-isomorphic-effect';


const randomId = () => `mantine-${Math.random().toString(36).slice(2, 11)}`;


const useReactId: () => string | undefined =
  (React as any)['useId'.toString()] || (() => undefined);


function useClientId() {
  const [uuid, setUuid] = useState('');


  useIsomorphicEffect(() => {
    setUuid(randomId());
  }, []);


  return uuid;
}


function getReactId() {
  const id = useReactId();
  return id ? `mantine-${id.replace(/:/g, '')}` : '';
}


export function useId(staticId?: string) {
  return typeof staticId === 'string' ? staticId : getReactId() || useClientId();
}
```